common/src/main/java/org/apache/celeborn/common/write/PushState.java

```diff
@@ -33,9 +35,12 @@ public class PushState {
   public AtomicReference<IOException> exception = new AtomicReference<>();
   private final InFlightRequestTracker inFlightRequestTracker;
 
+  private Set<PushFailedBatch> failedBatchSet;
+
   public PushState(CelebornConf conf) {
     pushBufferMaxSize = conf.clientPushBufferMaxSize();
     inFlightRequestTracker = new InFlightRequestTracker(conf, this);
+    failedBatchSet = new HashSet<>();
```

Comment:
Seems should use ConcurrentHashSet here to handle concurrent adding

Reply:
Get, fixed